### DISCUSSION TITLE
Additional accelerometers (code hygiene)

### DISCUSSION THREAD

**Discussion by dhskinner** [meshtastic/firmware#4614] [Q&A]
Hi - currently there a couple of accelerometers and motion sensors supported by Meshtastic that are all in one file at [.../src/AccelerometerThread.h](https://github.com/meshtastic/firmware/blob/master/src/AccelerometerThread.h) which makes it tricky to add new ones

Can I suggest from a code hygiene perspective it could be the right time to move AccelerometerThread.h into its own folder and tease apart the code into individual classes per sensor - like:

```
/src
    |->motion
        |-> AccelerometerThread.h
        |->MPU6050.cpp
        |->MPU6050.h
        |->...etc...
```

The reason for raising this is I'll shortly add a PR to add the ICM-20948 (n.b. the newer big brother from the MPU-6050) but it will need a couple of new code and header files to fully enable the onboard 9-Axis digital motion processor which is quite a cool... it would be a bit odd to add a bunch of new files into the [.../src/](https://github.com/meshtastic/firmware/tree/master/src) directory

Am more than happy to do the work and submit a small PR first for some light refactoring, but.... before I start I wanted to check in - will this annoy our admins by adding new folders and moving a bit of code around or is it do-able?

[edit - as of today (Sep 3rd) I did a scan of all open PR's and there doesn't seem to be any I could spot where anyone else is making changes to AccelerometerThread.h in parallel]

**Comment by jp-bennett** [comment]
> before I start I wanted to check in - will this annoy our admins by adding new folders and moving a bit of code around or is it do-able?

This is a @thebentern question ultimately, but I like the sound of this.

**Comment by dhskinner** [comment]
This was transferred to Issue [#4662](https://github.com/meshtastic/firmware/issues/4662) so can be closed
